setup.py

```diff
@@ -29,58 +29,140 @@ def parse_requirements(filename):
+    try:
+        import pkg_resources
+    except ImportError:
+        pass
+
+    print(
+        """
+    =============================DEBUG ASSISTANCE=============================
+    If you are seeing a compilation error please try the following steps to
+    successfully install cryptography:
+    1) Upgrade to the latest pip and try again. This will fix errors for most
+       users. See: https://pip.pypa.io/en/stable/installing/#upgrading-pip
+    2) If on Linux, with glibc < 2.12, you can set PIP_ONLY_BINARY=cmake in
+       order to retrieve the last manylinux1 compatible wheel.
+    3) If on Linux, with glibc < 2.12, you can cap "cmake<3.23" in your
+       requirements in order to retrieve the last manylinux1 compatible wheel.
+    4) Open an issue with the debug information that follows at
+       https://github.com/scikit-build/cmake-python-distributions/issues
+    """
+    )
+    print("    Python: %s" % '.'.join(str(v) for v in sys.version_info[:3]))
+    print("    platform: %s" % platform.platform())
+    if sys.platform.startswith("linux"):
+        try:
+            print("    glibc: %s" % os.confstr("CS_GNU_LIBC_VERSION"))
+        except:
+            try:
+                import ctypes
+                process_namespace = ctypes.CDLL(None)
+                gnu_get_libc_version = process_namespace.gnu_get_libc_version
+                gnu_get_libc_version.restype = ctypes.c_char_p
+                glibc_version = gnu_get_libc_version()
+                if not isinstance(glibc_version, str):
+                    glibc_version = glibc_version.decode("ascii")
+                    print("    glibc: %s" % glibc_version)
+            except:
+                pass
+    if sys.platform.startswith("darwin"):
+        try:
+            macos_ver = subprocess.check_output(
+                [
+                    sys.executable,
+                    "-sS",
+                    "-c",
+                    "import platform; print(platform.mac_ver()[0])",
+                ],
+                universal_newlines=True,
+                env={"SYSTEM_VERSION_COMPAT": "0"},
+            ).strip()
+            print("    macos: %s" % macos_ver)
+        except:
+            try:
+                print("    macos: %s" % platform.mac_ver()[0])
+            except:
+                pass
+    print("    machine: %s" % platform.machine())
+    print("    bits: %d" % (64 if sys.maxsize > 2**32 else 32))
+    for dist in ["pip", "setuptools", "scikit-build"]:
+        try:
+            version = pkg_resources.get_distribution(dist).version
+        except pkg_resources.DistributionNotFound:
+            version = "n/a"
+        print("    {}: {}".format(dist, version))
+    for key in ["PEP517_BUILD_BACKEND"]:
+        if key in os.environ:
+            print("    {}={}".format(key, os.environ[key]))
+    print(
+        """\
+    =============================DEBUG ASSISTANCE=============================
+    """
+    )
+    raise
```

Comment:
This mostly comes from `cryptography` with a bit of rework on the information being dumped.